install.sh

```diff
@@ -30,16 +30,61 @@ RED='\e[1;91m';    # Use Light Red for errors.
 YELLOW='\e[1;33m'; # Use Yellow for informational messages.
 GREEN='\e[1;32m';  # Use Green for success messages.
 BLUE='\e[1;34m';   # Use Blue for intrafunction messages.
-GRAY='\e[1;37m';   # Use Gray for program output.
+GRAY='\e[0;37m';   # Use Gray for program output.
 MAGENTA='\e[1;35m';
 RESET='\e[0m'
 
-echo -e "${GREEN}Welcome to Chromebrew!${RESET}\n"
+# skip all checks if running on a docker container
+grep ':/docker' /proc/self/cgroup &> /dev/null && CREW_FORCE_INSTALL=1
+
+# reject crostini
+if [[ -d /opt/google/cros-containers && "${CREW_FORCE_INSTALL}" != '1' ]]; then
+  echo -e "${RED}Crostini containers are not supported by Chromebrew :/${RESET}"
+  echo -e "${YELLOW}Run 'curl -Ls git.io/vddgY | CREW_FORCE_INSTALL=1 bash' to perform install anyway${RESET}"
+  exit 1
+fi
+
+# disallow non-stable channels Chrome OS
+if [ -f /etc/lsb-release ]; then
+  if [[ ! "$(< /etc/lsb-release)" =~ CHROMEOS_RELEASE_TRACK=stable-channel$'\n' && "${CREW_FORCE_INSTALL}" != '1' ]]; then
+    echo -e "${YELLOW}The beta, dev, and canary channel are unsupported by Chromebrew${RESET}"
+    echo -e "${YELLOW}Run 'curl -Ls git.io/vddgY | CREW_FORCE_INSTALL=1 bash' to perform install anyway${RESET}"
+    exit 1
+  fi
+else
+  echo -e "${YELLOW}Unable to detect system information, installation will continue.${RESET}"
+fi
+
+echo -e "\n${GREEN}Welcome to Chromebrew!${RESET}\n"
 if [ "${EUID}" == "0" ]; then
   echo -e "${RED}Chromebrew should not be installed or run as root.${RESET}"
   exit 1;
 fi
 
+# prompt user to enter the sudo password if it set
+# if the PASSWD_FILE specified by chromeos-setdevpasswd exist, that means a sudo password is set
+if [[ "$(< /usr/sbin/chromeos-setdevpasswd)" =~ PASSWD_FILE=\'([^\']+) ]] && [ -f "${BASH_REMATCH[1]}" ]; then
+  echo -e "${BLUE}Please enter the developer mode password${RESET}"
+  # reset sudo timeout
+  sudo -k
+  sudo /bin/true
+fi
+
+# force curl to use system libraries
+function curl () {
+  # retry if download failed
+  # the --retry/--retry-all-errors parameter in curl will not work with the 'curl: (7) Couldn't connect to server'
+  # error, a for loop is used here
+  for (( i = 0; i < 4; i++ )); do
+    env LD_LIBRARY_PATH='' ${CURL} --ssl -C - "${@}" && \
+      return 0 || \
+      echo -e "${YELLOW}Retrying, $((3-$i)) retries left.${RESET}"
+  done
```

Comment:
Just so you know, this will be 4 retries instead of 3.

Reply:
It means 1 initial try, then 3 retries😅

Reply:
The initial try is still a retry. ;)

Reply:
If we're defining a curl function here, why do we need the `CURL` environment variable?

Reply:
OK, I think I understand what is going on here.  The `CURL` environment variable is used inside this function.